module/mfrc522_parser.c

```diff
@@ -0,0 +1,177 @@
+			return &commands[i];
+
+	return NULL;
+}
+
+/**
+ * Parse a command with multiple arguments
+ *
+ * @param cmd Command struct to fill up
+ * @param input Remaining user input, mutable and allocated via kmalloc() in mfrc522_parse()
+ * @param ref_cmd Reference command returned by find_cmd_from_token()
+ *
+ * @return 0 on success, a negative number otherwise
+ */
+static int parse_multi_arg(struct mfrc522_command *cmd, char *input,
+			   const struct driver_command *ref_cmd)
+{
+	// We only enter this function if arguments have been given to the input
+	u8 parameter_amount = 0;
+	char *token;
+	char *extra_data;
+	u8 extra_data_len;
+	int ret;
+
+	while (input) {
+		token = strsep(&input, MFRC522_SEPARATOR);
+		parameter_amount++;
+
+		// The first parameter is the length of the extra data
+		if (parameter_amount == 1) {
+			ret = kstrtou8(token, 10, &extra_data_len);
+			if (ret == -EINVAL) {
+				pr_err("[MFRC522] Invalid parameter for Data length: Expected number but got %s\n",
+				       token);
+				return ret;
+			}
+
+			if (ret == -ERANGE) {
+				pr_err("[MFRC522] Invalid parameter for Data length: Expected Unsigned Byte (0 - 255) but got %s\n",
+				       token);
+				return ret;
+			}
+
+			if (extra_data_len > MFRC522_MAX_DATA_LEN) {
+				pr_err("[MFRC522] Invalid parameter for Data length: Length %d is too important (max length: 25)\n",
+				       extra_data_len);
+				return -1;
+			}
+		} else if (parameter_amount == 2) {
+			// The second parameter is the extra data
+			extra_data = token;
+		} else {
+			// Any parameters after the second one is an error
+			pr_err("[MFRC522] Invalid number of parameters: Got at least 3, which is more than the maximum of %d",
+			       MFRC522_MAX_PARAMETER_AMOUNT);
+			return -1;
+		}
+	}
+
+	pr_info("[MFRC522] Found %d parameters for command %s\n",
+		parameter_amount, ref_cmd->input);
+
+	if (parameter_amount != ref_cmd->parameter_amount) {
+		pr_err("[MFRC522] Invalid command: %s: Expected %d arguments but got %d\n",
+		       ref_cmd->input, ref_cmd->parameter_amount,
+		       parameter_amount);
+		return -2;
+	}
+
+	return mfrc522_command_init(cmd, ref_cmd->cmd, extra_data,
+				    extra_data_len);
+}
+
+int mfrc522_parse(struct mfrc522_command *cmd, const char *input, size_t len)
+{
+	// Inputs are always organized according to the same format
+	//
+	// <cmd>[:<extra_data_len>:<extra_data>]
+	//
+	// Therefore, we will either get a simple command without any colons (in which case
+	// we will only be able to run strsep on the string ONCE) or a command with colons
+	// in which case it is required for the length of the extra data to be present.
+	// Also, since strsep modifies its input, we need to create a copy of the user's
+	// input.
+
+	// The input's length does not account for a terminating NULL, but `strsep` expects
+	// it. Allocate one more byte for the NULL terminator
+	char input_cpy[MFRC522_MAX_INPUT_LEN + 1] = { 0 };
+	char *input_mut = &input_cpy[0];
```

Comment:
why not:
```suggestion
	char *input_mut = input_cpy;
```
?

Reply:
Cause I'm dumb :D I didn't re-read what I wrote